From 527edf2f2240f33df3c5a946185a1db771430eed Mon Sep 17 00:00:00 2001
From: Peter Griffin <peter.griffin@linaro.org>
Date: Fri, 15 Mar 2019 15:44:06 +0100
Subject: [PATCH 1/2] media: default_decoder_factory: create GpuVideoDecoder if
 mojo disabled

Despite chrome://gpu showing video acceleration is enabled, the
GpuVideoDecoder never actually gets created without this change
applied.

With this patch applied, VDA is created and used when playing back
video.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>
---
 media/renderers/default_decoder_factory.cc | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/media/renderers/default_decoder_factory.cc b/media/renderers/default_decoder_factory.cc
index 1229941..a71efb7 100644
--- a/media/renderers/default_decoder_factory.cc
+++ b/media/renderers/default_decoder_factory.cc
@@ -86,7 +86,6 @@ void DefaultDecoderFactory::CreateVideoDecoders(
   video_decoders->push_back(
       std::make_unique<DecryptingVideoDecoder>(task_runner, media_log));
 #endif
-
   // Perfer an external decoder since one will only exist if it is hardware
   // accelerated.
   // Remember that |gpu_factories| will be null if HW video decode is turned
@@ -97,14 +96,12 @@ void DefaultDecoderFactory::CreateVideoDecoders(
     // factories, require that their message loops are identical.
     DCHECK_EQ(gpu_factories->GetTaskRunner(), task_runner);
 
-    if (external_decoder_factory_) {
+    // MojoVideoDecoder replaces any VDA for this platform when it's enabled.
+    if (external_decoder_factory_ && base::FeatureList::IsEnabled(media::kMojoVideoDecoder)) {
       external_decoder_factory_->CreateVideoDecoders(
           task_runner, gpu_factories, media_log, request_overlay_info_cb,
           target_color_space, video_decoders);
-    }
-
-    // MojoVideoDecoder replaces any VDA for this platform when it's enabled.
-    if (!base::FeatureList::IsEnabled(media::kMojoVideoDecoder)) {
+    } else {
       video_decoders->push_back(std::make_unique<GpuVideoDecoder>(
           gpu_factories, request_overlay_info_cb, target_color_space,
           media_log));
-- 
2.7.4

